File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> clothMap = new HashMap<>();
        
        for (int i = 0; i < clothes.length; i++) {
            clothMap.put(clothes[i][1], clothMap.getOrDefault(clothes[i][1], 0) + 1);
        }
        
        for (String key : clothMap.keySet()) {
            answer *= (clothMap.get(key) + 1);
        }
        
        return answer - 1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    var answer = 1;
    let map = new Map();
    
    for ([name, type] of clothes) {
        map.set(type, (map.get(type) || 0) + 1);
    }
    
    for ([type, cnt] of map) {
        answer *= (cnt + 1);
    }
    
    return answer - 1;
}